PosteAPI/views.py

```diff
@@ -334,6 +339,67 @@ def post(self, request):
             # error message contained in response.data
             return response
 
+    def patch(self, request):
+        """
+        Edits a post in the server
+        :param request: Request object with post id and auth in header
+                        Request Body contains new title, description and url
+        """
+        #print(request)
+        try:
+            post_id = request.headers.get('id')
+            post = Post.objects.get(pk=post_id)
+            data = json.loads(request.body.decode('utf-8'))
+            post.edit(data.get('title'),data.get('description'),data.get('url'))
+            post.save()
```

Comment:
Would recommend moving `post.save()` into the edit method; otherwise, calling that method elsewhere will (perhaps unexpectedly) not save the changes.

Alternatively, this might be a good time to bring up django's built-in `update()` method; it doesn't *directly* apply here (as update() requires a QuerySet, whereas get() returns a single object) but maybe a good learning opportunity:

```
The update() method is applied instantly and returns the number of rows matched by the query (which may not be equal to the number of rows updated if some rows already have the new value). The only restriction on the [QuerySet](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#django.db.models.query.QuerySet) being updated is that it can only access one database table: the model’s main table. You can filter based on related fields, but you can only update columns in the model’s main table.
```

Update is neat because it is FAST (handles the queryset directly, allowing bulk-updates without retrieving individual objects, and which might be a performance consideration with many users) but, it has its drawbacks. Normally, creating or saving a model calls that models `save()` method (which you can override, to implement custom functionality). An `update()` bypasses the `save()` method, so I have mixed feelings on it (since you might expect it to do something in `save()` that is then skipped. Like I said, though, its fast, and you can do neat things like:
`Post.objects.filter(<some criteria>).update(<what you're updating>)`

Also, it doesn't work directly with get(). Oh well.

Reply:
Actually, I was editting and then realized my changes weren't being saved. And then my sleep deprived brain didn't think to put the save() method call in the edit function, rather in the actual body of the code. 

Im not fully sure about how to use update() yet, but down the line for longer changes we might be using it. Definitely something to learn.